content/source/docs/enterprise/sentinel/import/tfstate.html.md

```diff
@@ -10,103 +10,526 @@ description: |-
+```
+
+* **Return Type:** A [module namespace](#namespace-module).
+
+The `module()` function in the [root namespace](#namespace-root) returns the
+[module namespace](#namespace-module) for a particular module address.
 
-The terraform version that made this state.
+The address must be a list and is the module address, split on the period (`.`),
+excluding the root module.
 
-### tfstate.resources
+Hence, a module with an address of simply `foo` (or `root.foo`) would be
+`["foo"]`, and a module within that (so address `foo.bar`) would be read as
+`["foo", "bar"]`.
 
-Resources returns a map of all the resources in the root module.
-This is identical to tfstate.module([]).resources.
+As an example, given the following module block:
 
-### tfstate.data
+```hcl
+module "foo" {
+  # ...
+}
+```
 
-Resources returns a map of all the data sources in the root module.
-This is identical to tfstate.module([]).data.
+If the module contained the following content:
 
-### tfstate.module(path)
+```hcl
+resource "null_resource" "foo" {
+  triggers = {
+    foo = "bar"
+  }
+}
+```
 
-The module function finds the module at the given path.
 
-## Type: m
+The following policy would evaluate to `true`, if the resource was present in
+the state:
 
-### m.path
+```python
+import "tfstate"
 
-This returns the path for this module. This is a usable argument to
-tfstate.module() if necessary.
+main = rule { tfstate.module(["foo"]).resources.null_resource.foo[0].attr.triggers.foo is "bar" }
+```
 
-### m.resources
+### Value: `module_paths`
 
-This returns all the resources in the module as a map. The map
-key is the type, the value is the same as `m.resources.TYPE`.
+* **Value Type:** List of a list of strings.
 
-### m.resources.TYPE
+The `module_paths` value within the [root namespace](#namespace-root) is a list
+of all of the modules within the Terraform state at plan-time.
 
-This returns all the resources in the module with the given type
-as a map from name to resource. For the documentation on the
-value, see `m.resources.TYPE.NAME`
+Note the distinction here - this means if a module is not present in state, it
+will not be present in `module_paths`.
 
-### m.resources.TYPE.NAME
+This data is represented as a list of a list of strings, with the inner list
+being the module address, split on the period (`.`).
 
-This returns the list of resources with the given type and name. The
-result is a list because the plan is aware that the count may be greater
-than zero.
+The root module is included in this list, represented as an empty inner list, as
+long as it is present in state.
 
-### m.data
+As an example, if the following module block was present within a Terraform
+configuration:
 
-This returns all the data sources in the module as a map. The map key is the
-type, the value is the same as `m.data.TYPE`.
+```hcl
+module "foo" {
+  # ...
+}
+```
 
-### m.data.TYPE
+The following policy would evaluate to `true`, _only_ if the diff had changes
+for that module:
```

Comment:
Hm, is this maybe copypasta? It seems like it should say "The following policy would evaluate to true only if the module is present in the state:"